module/VuFind/src/VuFind/Search/Factory/AbstractSolrBackendFactory.php

```diff
@@ -380,6 +380,31 @@ protected function createConnector()
                 $this->serviceLocator->get(\VuFindHttp\HttpService::class)
             );
         }
+
+        if (!empty($config->IndexCache->adapter)) {
+            $cacheConfig = $config->IndexCache->toArray();
+            $options = $cacheConfig['options'] ?? [];
+            if (empty($options['namespace'])) {
+                $options['namespace'] = 'Index';
+            }
+            if (empty($options['ttl'])) {
+                $options['ttl'] = 300;
+            }
+            if ('Memcached' === $cacheConfig['adapter']) {
+                $servers = $cacheConfig['servers'] ?? ['localhost:11211'];
+                foreach ($servers as $server) {
+                    $options['servers'][] = explode(':', $server, 2);
+                }
+            }
+            $settings = [
+                'adapter' => [
+                    'name' => $cacheConfig['adapter'],
+                    'options' => $options,
+                ]
+            ];
+            $cache = \Laminas\Cache\StorageFactory::factory($settings);
```

Comment:
Did you consider leveraging the `\VuFind\Cache\Manager` here instead of directly hitting the `\Laminas\Cache\StorageFactory`? I'm not necessarily saying that would be a better solution -- honestly, I think VuFind's Cache Manager could use some work -- but I'm wondering if it makes sense to centralize the logic, and whether this might be an opportunity to make some improvements to caching more generally.

Reply:
I did, but since it's pretty much focused on caching configuration etc. in files, it didn't feel like it would make sense. Also, I believe it makes sense to keep configuration such as ttl separate.

Reply:
Makes sense. I think we should revisit the whole thing in future, but it seems most likely that we'll eliminate/simplify the cache manager in future, if we change anything at all, so I think it does make sense to keep this separate for now.